app/components/ToolPanel/ToolPanel.tsx

```diff
@@ -133,25 +168,57 @@ const ToolPanel = () => {
           </div>
 
           <button
-            onClick={() => {
-              setActiveLayer({
-                daily: activeLayers.daily,
-                weekly: !activeLayers.weekly,
-              });
-            }}
+            onClick={toggleWeeklyLayer}
             className={`flex items-center gap-2 p-2 ${
-              activeLayers.weekly ? 'bg-white' : 'bg-gray-200'
+              isWeeklyyActive ? 'bg-white' : 'bg-gray-200'
             } rounded-md`}
           >
             Weekly
-            {activeLayers.weekly ? <FaEye /> : <FaEyeSlash />}
+            {isWeeklyyActive ? <FaEye /> : <FaEyeSlash />}
           </button>
         </div>
-        {activeLayers.weekly && (
-          <div className="py-2">
-            <p className="text-sm">
-              Weekly events are those of 2.5 or greater magnitude
-            </p>
+        {isWeeklyyActive && (
+          <div className="flex gap-2 py-3">
+            <fieldset>
+              <legend>Mag: </legend>
+            </fieldset>
+
+            <div>
+              <input
+                type="checkbox"
+                id="high"
+                name="high"
+                value="high"
+                checked={activeLayers.weekly.high}
+                onChange={() => toggleWeeklyMagnitude('high')}
+                style={{ marginRight: '4px' }}
+              />
+              <label htmlFor="high">High</label>
```

Comment:
All of these inputs could be abstracted into a reusable component for the daily and weekly layers